src/languages/clojure.js

```diff
@@ -45,20 +45,44 @@ export default function(hljs) {
       'lazy-seq spread list* str find-keyword keyword symbol gensym force rationalize'
   };
 
-  const SIMPLE_NUMBER_RE = '[-+]?\\d+(\\.\\d+)?';
-
   const SYMBOL = {
     begin: SYMBOL_RE,
     relevance: 0
   };
   const NUMBER = {
-    className: 'number',
-    begin: SIMPLE_NUMBER_RE,
-    relevance: 0
+    scope: 'number',
+    relevance: 0,
+    variants: [
+      {match: /[-+]?0[xX][0-9a-fA-F]+N?/},                           // hexadecimal                 // 0x2a
+      {match: /[-+]?0[0-7]+N?/},                                     // octal                       // 052
+      {match: /[-+]?[1-9][0-9]?[rR][0-9a-zA-Z]+N?/},                 // variable radix from 2 to 36 // 2r101010, 8r52, 36r16
+      {match: /[-+]?[0-9]+\/[0-9]+N?/},                              // ratio                       // 1/2
+      {match: /[-+]?[0-9]+((\.[0-9]*([eE][+-]?[0-9]+)?M?)|([eE][+-]?[0-9]+M?|M))/}, // float        // 0.42 4.2E-1M 42E1 42M
+      {match: /[-+]?([1-9][0-9]*|0)N?/},                             // int (don't match leading 0) // 42 42N
+    ]
   };
+  const CHARACTER = {
+    scope: 'character',
+    variants: [
+      {match: /\\o[0-3]?[0-7]{1,2}/},                             // Unicode Octal 0 - 377
+      {match: /\\u[0-9a-fA-F]{4}/},                               // Unicode Hex 0000 - FFFF
+      {match: /\\(newline|space|tab|formfeed|backspace|return)/}, // special characters
+      {match: /\\\S/}                                             // any non-whitespace char
```

Comment:
This is so tiny a match... can we do any looking forward to qualify it?  I'm imagining false positives with any language that allows `\SOMETHING` style tags...  if we can't qualify it I think we should make it `relevance: 0`.

Reply:
It is a literal and can appear anywhere, so you are right with `relevance: 0`. 
I didn't put any thought in this or the other modes, what the `relevance` may be. What is "typically Clojure"?

It would be a nice exercise to generate possible values for modes and try to match them against other languages to automatically calculate the relevance. 🤔

Reply:
I added `relevance: 0`.

Although, I am not sure how many programming languages support the full first Unicode plane. `\⠓` is a valid Clojure literal.